src/poetry/utils/env/python_manager.py

```diff
@@ -1,52 +1,88 @@
 from __future__ import annotations
 
 import shutil
-import subprocess
 import sys
 
 from functools import cached_property
 from pathlib import Path
 from typing import TYPE_CHECKING
+from typing import cast
+from typing import overload
+
+import findpython
+import packaging.version
 
 from cleo.io.null_io import NullIO
 from cleo.io.outputs.output import Verbosity
 from poetry.core.constraints.version import Version
-from poetry.core.constraints.version import parse_constraint
 
-from poetry.utils._compat import decode
 from poetry.utils.env.exceptions import NoCompatiblePythonVersionFoundError
-from poetry.utils.env.script_strings import GET_PYTHON_VERSION_ONELINER
 
 
 if TYPE_CHECKING:
+    from collections.abc import Iterable
+
     from cleo.io.io import IO
+    from typing_extensions import Self
 
     from poetry.config.config import Config
     from poetry.poetry import Poetry
 
 
+class ShutilWhichPythonProvider(findpython.BaseProvider):  # type: ignore[misc]
+    @classmethod
+    def create(cls) -> Self | None:
+        return cls()
+
+    def find_pythons(self) -> Iterable[findpython.PythonVersion]:
+        if path := shutil.which("python"):
+            return [findpython.PythonVersion(executable=Path(path))]
+        return []
+
+
 class Python:
-    def __init__(self, executable: str | Path, version: Version | None = None) -> None:
-        self.executable = Path(executable)
-        self._version = version
+    @overload
+    def __init__(self, *, python: findpython.PythonVersion) -> None: ...
+
+    @overload
+    def __init__(
+        self, executable: str | Path, version: Version | None = None
+    ) -> None: ...
+
+    # we overload __init__ to ensure we do not break any downstream plugins
+    # that use the this
+    def __init__(
+        self,
+        executable: str | Path | None = None,
+        version: Version | None = None,
+        python: findpython.PythonVersion | None = None,
+    ) -> None:
```

Comment:
I would like to keep `findpython` as an implementation detail of the `Python` class, therefor we should avoid `findpython` in the public "API" (like here in `__init__()`).

As far as I can see passing `python` on init is only done by several classmethods. Those method could pass the executable instead and the `PythonVersion` object is then (re)created. (Yes, recreating might be some overhead but it shouldn't be that much and for the sake of separating the implementation details I clearly prefer that). Not sure if the version should than also be passed through, because it looks like it's lazy loaded in `PythonVersion`.

Reply:
It will trigger unnecessary subprocess overheads and is not recommended. I don't think there is anything wrong with using it as an init var.

Reply:
Resolving this, I do not think I want to remove the ability to pass in an already instantiated `PythonVersion` instance.